crates/red_knot_python_semantic/src/types.rs

```diff
@@ -93,6 +100,24 @@ pub(crate) fn global_symbol_ty<'db>(db: &'db dyn Db, file: File, name: &str) ->
     symbol_ty(db, global_scope(db, file), name)
 }
 
+/// Shorthand for `symbol_ty_by_id` that looks up a global symbol from the context of being in that
+/// module.
+pub(crate) fn global_symbol_lookup<'db>(db: &'db dyn Db, file: File, name: &str) -> Type<'db> {
```

Comment:
I still think it might be fine to inline this into `TypeInferenceBuilder::lookup_name`, since I don't think it will have any other call sites? If we do keep it as a separate function, then I think it should include the builtins-fallback part too; that's part of a "global symbol lookup".

Also I don't think it needs to be `pub(crate)`.